src/CreativeCommons/CreativeCommons.tsx

```diff
@@ -0,0 +1,221 @@
+      </svg>
+    ),
+  },
+});
+
+const licenses = {
+  attr: 'Attribution 4.0 International',
+  attrNoDeriv: 'Attribution-NoDerivatives 4.0 International',
+  attrShareAlike: 'Attribution-ShareAlike 4.0 International',
+  attrNC: 'Attribution-NonCommercial 4.0 International',
+  attrNoDerivNC: 'Attribution-NonCommercial-NoDerivatives 4.0 International',
+  attrShareAlikeNC: 'Attribution-NonCommercial-ShareAlike 4.0 International',
+  cc0: 'CC0 1.0 Universal',
+};
+
+const CCIcon: FC<CCIconProps> = ({ icon, title, description }) => {
+  const tooltip = (
+    <>
+      <Typography fontWeight='bold'>{title}</Typography>
+      {description && <Typography variant='body2'>{description}</Typography>}
+    </>
+  );
+
+  return (
+    <div>
+      <Tooltip id={title} title={tooltip} arrow>
+        {icon}
+      </Tooltip>
+    </div>
+  );
+};
+
+type CreativeCommonsProps = {
+  requireAccreditation?: boolean | undefined;
+  allowSharedAdaptation: CCSharingVariant;
+  allowCommercialUse: boolean;
+  iconSize?: number | undefined;
+  withLicenseName?: boolean | undefined;
+  textColor?: string | undefined;
+  textSize?: number | undefined;
+  sx?: SxProps;
+};
+
+const CreativeCommons: FC<CreativeCommonsProps> = (props) => {
+  const {
+    requireAccreditation = true,
+    allowCommercialUse,
+    allowSharedAdaptation,
+    iconSize = 50,
+    withLicenseName = true,
+    sx,
+    textColor = PRIMARY_COLOR,
+    textSize = Math.max(iconSize / 4, 10),
+  } = props;
+
+  const iconData = React.useMemo(() => ccData(iconSize), [iconSize]);
+
+  const additionalIcons = requireAccreditation ? (
+    <>
+      <CCIcon {...iconData.by} />
+      {!allowCommercialUse && <CCIcon {...iconData.nc} />}
+      {allowSharedAdaptation === 'no' && <CCIcon {...iconData.nd} />}
+      {allowSharedAdaptation === 'alike' && <CCIcon {...iconData.sa} />}
+    </>
+  ) : (
+    <CCIcon {...iconData.cc0} />
+  );
+
+  console.log(allowSharedAdaptation);
+
+  const license = requireAccreditation
+    ? allowCommercialUse
+      ? allowSharedAdaptation === 'yes'
+        ? licenses.attr
+        : allowSharedAdaptation === 'no'
+        ? licenses.attrNoDeriv
+        : licenses.attrShareAlike
+      : allowSharedAdaptation === 'yes'
+      ? licenses.attrNC
+      : allowSharedAdaptation === 'no'
+      ? licenses.attrNoDerivNC
+      : licenses.attrShareAlikeNC
+    : licenses.cc0;
```

Comment:
Nitpick: could this be factored out as a helper function in SDK? It would return the license name or abbreviation (or both) given the conditions